contribs/src/main/java/com/netflix/conductor/contribs/lock/LocalOnlyLock.java

```diff
@@ -75,33 +79,47 @@ public boolean acquireLock(String lockId, long timeToTry, long leaseTime, TimeUn
                 unit);
         if (acquireLock(lockId, timeToTry, unit)) {
             LOGGER.trace("Releasing {} automatically after {} {}", lockId, leaseTime, unit);
-            SCHEDULER.schedule(() -> releaseLock(lockId), leaseTime, unit);
+            scheduledFutures.put(
+                    lockId, SCHEDULER.schedule(() -> releaseLock(lockId), leaseTime, unit));
             return true;
         }
         return false;
     }
 
+    private void removeLeaseExpirationJob(String lockId) {
+        ScheduledFuture<?> schedFuture = scheduledFutures.get(lockId);
+        if (schedFuture != null && schedFuture.cancel(false)) {
+            scheduledFutures.remove(lockId);
+            LOGGER.trace("lockId {} removed from lease expiration job", lockId);
+        }
+    }
+
     @Override
     public void releaseLock(String lockId) {
         // Synchronized to prevent race condition between semaphore check and actual release
         // The check is here to prevent semaphore getting above 1
         // e.g. in case when lease runs out but release is also called
-        synchronized (CACHE) {
-            if (CACHE.getUnchecked(lockId).availablePermits() == 0) {
+        synchronized (lockIdtoSemaphoreMap) {
+            if (lockIdtoSemaphoreMap.getUnchecked(lockId).availablePermits() == 0) {
                 LOGGER.trace("Releasing {}", lockId);
-                CACHE.getUnchecked(lockId).release();
+                lockIdtoSemaphoreMap.getUnchecked(lockId).release();
+                removeLeaseExpirationJob(lockId);
             }
         }
     }
 
     @Override
     public void deleteLock(String lockId) {
         LOGGER.trace("Deleting {}", lockId);
-        CACHE.invalidate(lockId);
+        lockIdtoSemaphoreMap.invalidate(lockId);
     }
 
     @VisibleForTesting
     LoadingCache<String, Semaphore> cache() {
-        return CACHE;
+        return lockIdtoSemaphoreMap;
+    }
+
+    ConcurrentHashMap<String, ScheduledFuture<?>> scheduledFutures() {
```

Comment:
if this in package scope for testing purposes, please annotate with `@VisibleForTesting`.